components/channel_layout/center_channel/index.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017 Mattermost Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+import {getTeamByName} from 'mattermost-redux/selectors/entities/teams';
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+
+import Constants from 'utils/constants';
+import {getGlobalItem} from 'selectors/storage';
+
+import CenterChannel from './center_channel';
+
+const getLastChannelPath = (state, teamName) => {
```

Comment:
Could probably turn this into a selector but I'm 0/5

Reply:
I don't know if that would change much since all it's doing is getting data without computing anything from it. It would also make the logic a bit more complicated since it needs to get the team first before it gets the channel

Reply:
Yeah. I had used reselect, but it got pretty complex and I didn't think there minimal perf gains because this is just getting data.